### ISSUE TITLE
[Enhancement]: Email template header image blurry/stretched

### ISSUE THREAD

**Issue by natko1** [woothemes/woocommerce#40080]
### Describe the solution you'd like

I believe this is a bug. Why is the header image by default set to max-width 100%. This unnecessarily stretches smaller images to conform with this requirement, and obviously, this causes image quality issues.

Why not have a setting to change this?

**Comment by barryhughes** [comment]
Hi @natko1,

I'd like to make sure we're seeing the same thing. I visited **WooCommerce ▸ Settings ▸ Emails** and provided the following URL for the header image (per the final query parameter, the image has a width of 320px):

```
https://unsplash.com/photos/6F2I3dPQ73U/download?ixid=M3wxMjA3fDB8MXxzZWFyY2h8N3x8ZHJ1aWRzfGVufDB8fHx8MTY5NDIxNjE4NXww&force=true&w=320
```

When I preview the image, this is what I see:

<img width="1131" alt="email-preview" src="https://github.com/woocommerce/woocommerce/assets/3594411/e61c1df9-9053-4b5f-982b-5993f33ed073">

So, the image has not been stretched, but rather is rendering with the correct aspect ratio and without exceeding its own width. Things that might be useful for us to know:

- Do you still see the problem when only WooCommerce is active, alongside a default theme?
- Can you provide a screenshot showing what you see?
- Are you seeing this with a specific email client (unfortunately, the reality is that not all email clients render things in quite the same way)?

**Comment by natko1** [comment]
Hi Barry, thanks for chiming in.

1. Yep
2. Sure, see attached.
3.
image file used
![EZ-Gift-Cards-Logo-Small](https://github.com/woocommerce/woocommerce/assets/4186392/7e43afcf-0974-4b46-8320-a133032bc80b)
![image](https://github.com/woocommerce/woocommerce/assets/4186392/67c94f7d-25ba-4028-a5b9-815bbaa9cc06)
result
![image 2](https://github.com/woocommerce/woocommerce/assets/4186392/8c81d936-d0bc-457c-b922-979452c5e7ac)
5. Nope, not related to any specific email client

**Comment by natko1** [comment]
@barryhughes there's no way to specify the exact dimensions you would like to use though for the email header image via a toggle or something similar?

As, on my end our logo appears blurry and stretched if the width is set below 1080px it seems. See attached.
![see here](https://github.com/woocommerce/woocommerce/assets/4186392/96d78bb6-1b3e-4fdb-afe7-ac5388131847)

Do you still see the problem when only WooCommerce is active, alongside a default theme? Yep
Are you seeing this with a specific email client: nope, experiencing it with all clients

**Comment by barryhughes** [comment]
OK. So are you viewing the email on a HiDPI ("Retina") display? That would to some extent explain what you are seeing, since a smaller image will become stretched and anti-aliased.

We can consider a size setting (or some other solution) to accommodate this. In the meantime, you could perhaps create a higher resolution version of your image, with empty/transparent padding, and use that to achieve a better fit and look?

**Comment by natko1** [comment]
Hi, well I am using an iPhone/MacBook and also a non Retina display Dell laptop and I experience the issue on all devices.

I have uploaded a 1080px image, however it looks a bit odd as you can see below. It's massive as it goes all the way from one end to the other. So it resolved the blurry/stretched issue, but is not really aesthetically pleasing to have such a big logo for emails.
![see here 2](https://github.com/woocommerce/woocommerce/assets/4186392/dfe62bf4-1011-4b2a-b48e-e0bd1aa854db)

> We can consider a size setting (or some other solution) to accommodate this.

That would be much appreciated!

P.S. MailPoet plugin has a good method for this and a simple width toggle which works flawlessly for scaling down larger images so you don't get that blurry/stretched issue. See below:
![mailpoet](https://github.com/woocommerce/woocommerce/assets/4186392/792b531c-20e2-4d20-9563-3d0797887134)
![mailpoet 2](https://github.com/woocommerce/woocommerce/assets/4186392/526e21f6-d148-4f51-9dbb-90531b83e489)

Looks like I'm not the only one who has this issue ([quick search](https://wordpress.org/support/topic/email-header-image-logo-blurry/)).

I see the low priority tag, but I think _brand image is pretty important for online businesses_, so hopefully it doesn’t take too long to implement. Having shoddy images sent to customers isn’t a good look. 🫢

**Comment by barryhughes** [comment]
> I have uploaded a 1080px image, however it looks a bit odd as you can see below. It's massive as it goes all the way from one end to the other.

Understood. To clarify, I was suggesting a larger scale image in which the logo itself is surrounded by an appropriate amount of (transparent) padding, something like this:

<div align="center"><img src="https://github.com/woocommerce/woocommerce/assets/3594411/332fdb00-2938-40d6-a748-80a1e1ae1cfc" width="500" /></div>

> MailPoet plugin has a good method for this and a simple width toggle which works flawlessly for scaling down larger images so you don't get that blurry/stretched issue.

That looks nice! 

> I see the low priority tag, but I think brand image is pretty important for online businesses, so hopefully it doesn’t take too long to implement.

Understood, but we cannot offer any guarantees. Especially in a case like this, where various workarounds are possible (such as crafting the header image as I described above, or exercising fine-grained control via template overrides), it isn't going to be top of our list.

**Comment by natko1** [comment]
> Understood. To clarify, I was suggesting a larger scale image in which the logo itself is surrounded by an appropriate amount of (transparent) padding, something like this:

Tried that, but the image is still not clear unfortunately.

> Understood, but we cannot offer any guarantees. Especially in a case like this, where various workarounds are possible (such as crafting the header image as I described above, or exercising fine-grained control via template overrides), it isn't going to be top of our list.

Okay, that's understandable, I guess. It just makes things a bit difficult for less tech savvy people such as myself. Tinkering with overrides and that. Then, if a email template gets updated, adjustments need to be made again.

Are we talking a year, two or more? I won't hold you to it (don't worry)!😉

**Comment by barryhughes** [comment]
I'll just add one other possibility for a temporary workaround, which is a code snippet looking something like this (it would likely need adjustment for specific cases):

```php
add_filter( 'woocommerce_email_styles', function ( $default_css ) {
	$additional_email_css = '#template_header_image img { width: 300px; }';
	return $default_css . PHP_EOL . $additional_email_css;
} );
```

> Are we talking a year, two or more? I won't hold you to it (don't worry)!

I'm afraid that's an unknown at this time.

**Comment by natko1** [comment]
Thank you. That code worked pretty good. 👍

**Comment by pavel-mailpoet** [comment]
Thank you for raising this enhancement request. We appreciate your input on header image handling.

We’re currently focused on building a new editor that will help with this issue. 

In the meantime, please use the CSS workaround shared above to set your header image width manually.

Adding a new template setting isn’t on our roadmap right now. We’ll close this issue but may revisit it once the editor work is complete.

Thanks again for contributing and for your understanding.
